### PR TITLE
[BEAM-10961] turn on enableStrictDependencies globally

### PR DESCRIPTION
**Pull request by shehzaadn-vd**
Now that most projects have been validated for strict dependency checking, we are turning on enableStrictDependencies globally, removing enableStrictDependencies:true from all projects that have been corrected (and necessary changes merged) so that they can inherit this global switch, and adding enableStrictDependencies:false for the small number of files which are still under review in other PRs (thus overriding the global switch).

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by shehzaadn-vd** [comment]
Run Java PreCommit

File: buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_26_0.groovy [hunk 1]
```diff
@@ -51,6 +51,7 @@ class GrpcVendoring_1_26_0 {
       "com.google.protobuf:protobuf-java:$protobuf_version",
       "com.google.protobuf:protobuf-java-util:$protobuf_version",
       "com.google.code.gson:gson:$gson_version",
+      "io.grpc:grpc-api:h$grpc_version",
```
Comment (ibzib) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_26_0.groovy:54]: Is this a typo?

```suggestion
      "io.grpc:grpc-api:$grpc_version",
```

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -177,7 +177,11 @@ class BeamModulePlugin implements Plugin<Project> {
      * By default, skips anything in the `org.apache.beam.model` namespace.
      */
     List<String> generatedClassPatterns = [
-      "^org\\.apache\\.beam\\.model.*"
+      "^org\\.apache\\.beam\\.model\\.pipeline\\.v1\\.*",
```
Comment (ibzib) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:180]: Doesn't `^org\\.apache\\.beam\\.model.*` cover all of the these except the windmill worker?

**Comment by shehzaadn-vd** [comment]
@ibzib Java Wordcount Direct Runner no MacOS is failing with:

`> Task :runners:core-construction-java:compileJava
warning: Memory constraints are impeding performance; please increase max heap size.
warning: Memory constraints are impeding performance; please increase max heap size.
error: warnings found and -Werror specified`

Suggests this will be a flappy check going forward. Shall we set the compiler JVM max heap size in BeamModulePlugin.groovy?

File: buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_26_0.groovy (hunk 1 above)
Comment (shehzaadn-vd) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_26_0.groovy]: Removed this line. Was unnecessary.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 2 above)
Comment (shehzaadn-vd) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Yes. Those changes weren’t necessary. Wasn’t even necessary to add windmill worker...

**Comment by shehzaadn-vd** [comment]
Run Java_Examples_Dataflow PreCommit

File: release/build.gradle.kts [hunk 3]
```diff
@@ -30,6 +30,7 @@ val library = project.extensions.extraProperties["library"] as Map<String, Map<S
 dependencies {
   compile(library.getValue("groovy").getValue("groovy_all"))
   compile("commons-cli:commons-cli:1.2")
+  permitUnusedDeclared("commons-cli:commons-cli:1.2") // BEAM-11761
```
Comment (shehzaadn-vd) [inline release/build.gradle.kts:33]: @ibzib do you think this is necessary for the "release" project?

Comment (ibzib) [reply release/build.gradle.kts:33]: Not sure, but permitUnusedDeclared is fine either way.

**Review by ibzib — APPROVED** [review]
LGTM

**Comment by ibzib** [comment]
@shehzaadn-vd Can you please squash the commit history?

**Comment by shehzaadn-vd** [comment]
> @shehzaadn-vd Can you please squash the commit history?

done!

**Comment by shehzaadn-vd** [comment]
Run Portable_Python PreCommit

**Pull request merged**